pyvista/core/dataset.py

```diff
@@ -242,11 +244,25 @@ def arrows(self) -> Optional['pyvista.PolyData']:
     @property
     def vectors(self) -> Optional[pyvista_ndarray]:
         """Return active vectors."""
+        warnings.warn(
+            "Use of `DataSet.vectors` is deprecated.  Use `DataSet.active_vectors` instead.",
+            PyvistaDeprecationWarning
+        )
         return self.active_vectors
 
     @vectors.setter
     def vectors(self, array: np.ndarray):
         """Set the active vector."""
+        warnings.warn(
+            "Use of `DataSet.vectors` to add vector data is deprecated. "
+            "Use `DataSet['vector_name'] = data` or "
+            "`DataSet.point_arrays['vector_name'] = data` or "
+            "`DataSet.cell_arrays['vector_name'] = data`. "
+            "Use `DataSet.set_active_vectors('vector_name')` or "
+            "`DataSet.active_vectors_name = 'vector_name' to make active."
+            ,
```

Comment:
This is terribly verbose. Do we need all this? I guess there's no good place to point users to in the warning in a concise manner?

And on a side note, when do we need to set `active_vectors_name`? Do we need it when there's only one set of vectors? Do we need it when there's an array called `'vectors'`?

Reply:
I agree looking at it here.  We can just use `DataSet['vector_name'] = data` and `DataSet.set_active_vectors('vector_name')`.  There is no need to list all the possibilites, just one possibility.

Reply:
> And on a side note, when do we need to set active_vectors_name? Do we need it when there's only one set of vectors? Do we need it when there's an array called 'vectors'?

I don't _think_ that "vectors" or "_vectors", after this PR, has a special meaning. The active_vectors designation is used in a few places, like in the arrows property (there is another issue here I will raise at some point), or streamlines generation.  In streamlines, the vtk filter uses the active vectors to generate the streamlines.  See:
https://github.com/pyvista/pyvista/blob/a9a6ee87098b1fcd5276615ebcad76836369900d/pyvista/core/filters/data_set.py#L2032-L2034

I'm not sure if vtkStreamTracer allows for using a non-active vector(s?).  For now, pyvista is using this in a few places.

Reply:
Yeah, thanks, I don't know what I remembered. I was probably just confused. In the meantime I also checked that `glyph` works with both `vector` and `vectors`. It's just that when there's a single set of vectors it's probably always the active one (I haven't checked).